src/System.Net.Security/src/System/Net/SecureProtocols/NegoState.Windows.cs

```diff
@@ -14,871 +13,607 @@
-            }
+            return (win32exception != null) && (win32exception.NativeErrorCode == (int)Interop.SecurityStatus.LogonDenied);
+        }
 
-            else if (protectionLevel == ProtectionLevel.EncryptAndSign)
-            {
-                flags |= Interop.SspiCli.ContextFlags.Confidentiality;
-            }
-            else if (protectionLevel == ProtectionLevel.Sign)
-            {
-                // Assuming user expects NT4 SP4 and above.
-                flags |= Interop.SspiCli.ContextFlags.ReplayDetect | Interop.SspiCli.ContextFlags.SequenceDetect | Interop.SspiCli.ContextFlags.InitIntegrity;
-            }
+        internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
+        {
+            return new Win32Exception((int)SslStreamPal.GetInteropFromSecurityStatusPal(statusCode));
+        }
+
+        internal static int Encrypt(
+            SafeDeleteContext securityContext,
+            byte[] buffer,
+            int offset,
+            int count,
+            bool isConfidential,
+            bool isNtlm,
+            ref byte[] output,
+            uint sequenceNumber)
+        {
+            SecSizes sizes = SSPIWrapper.QueryContextAttributes(
+                GlobalSSPI.SSPIAuth,
+                securityContext,
+                Interop.SspiCli.ContextAttribute.Sizes
+                ) as SecSizes;
 
-            if (isServer)
+            try
             {
-                if (_extendedProtectionPolicy.PolicyEnforcement == PolicyEnforcement.WhenSupported)
-                {
-                    flags |= Interop.SspiCli.ContextFlags.AllowMissingBindings;
-                }
+                int maxCount = checked(Int32.MaxValue - 4 - sizes.BlockSize - sizes.SecurityTrailer);
 
-                if (_extendedProtectionPolicy.PolicyEnforcement != PolicyEnforcement.Never &&
-                    _extendedProtectionPolicy.ProtectionScenario == ProtectionScenario.TrustedProxy)
+                if (count > maxCount || count < 0)
                 {
-                    flags |= Interop.SspiCli.ContextFlags.ProxyBindings;
+                    throw new ArgumentOutOfRangeException("count", SR.Format(SR.net_io_out_range, maxCount));
                 }
             }
-            else
+            catch (Exception e)
             {
-                // Server side should not request any of these flags.
-                if (protectionLevel != ProtectionLevel.None)
-                {
-                    flags |= Interop.SspiCli.ContextFlags.MutualAuth;
-                }
-
-                if (impersonationLevel == TokenImpersonationLevel.Identification)
+                if (!ExceptionCheck.IsFatal(e) && GlobalLog.IsEnabled)
                 {
-                    flags |= Interop.SspiCli.ContextFlags.InitIdentify;
+                    GlobalLog.Assert("NTAuthentication#" + LoggingHash.HashString(securityContext) + "::Encrypt", "Arguments out of range.");
```

Comment:
I'm concerned these files aren't up-to-date.  I previously went through and added Debug.Fail calls next to such asserts, but I don't see one here.  It's of course possible I missed one, but that makes me think that this code being refactored is actually an old copy, which means we could easily be losing a lot more changes in the process.